ksql-common/src/main/java/io/confluent/ksql/util/KsqlConfig.java

```diff
@@ -187,6 +187,13 @@
   public static final Collection<CompatibilityBreakingConfigDef> COMPATIBLY_BREAKING_CONFIG_DEFS
       = ImmutableList.of();
 
+  public static final String KSQL_SHUTDOWN_TIMEOUT_MS_CONFIG =
+      "ksql.streams.shutdown.timeout.ms";
+  public static final Long KSQL_SHUTDOWN_TIMEOUT_MS_DEFAULT = 300_000L;
```

Comment:
I thought to go with a more conservative default of 5 minutes made sense, to make sure everything has time to shut down properly.  Let me know if another value is preferred.